scripts/ladder.py

```diff
@@ -0,0 +1,436 @@
+    if args.s3:
+        permanent_data_prefix = "s3://ai2-llm"
+    permanent_data_prefix.rstrip("/")
+
+    remote_save_folder = f"s3://ai2-llm/checkpoints/OLMo-ladder/{run_name}"
+    load_path = args.load_path
+    if load_path is None:
+        load_path = find_latest_checkpoint(remote_save_folder)
+
+    model_config = MODEL_CONFIGS[args.model]
+    model_size = parse_size(args.model)
+    length_in_tokens = parse_length(args.length, model_size)
+
+    # calculate batch size according to
+    # https://www.semanticscholar.org/reader/5585191b1b479346ecf173be3b35c8313b77d457
+    # holds only for a sequence length of 2048 (but could probably be easily adapted)
+    assert model_config.max_sequence_length == 2048
+    global_batch_size = 160 * (model_size / 108000000) ** (2 / 3)
+    global_batch_size /= 8 * 4  # 8 GPUs per node, microbatch size 4
+    global_batch_size = round(global_batch_size)
+    global_batch_size *= 8 * 4
+
+    # We don't want the global batch size depend on the device batch size, because we might have to change the
+    # device batch size based on the hardware we're running on.
+    device_batch_size = {
+        "150M": 32,
+        "300M": 10,
+        "750M": 8,
+        "1B": 2,
+        "7B": 2,
+    }.get(args.model, 4)
+
+    assert global_batch_size % device_batch_size == 0
+
+    # calculate the learning rate according to the same paper
+    lr = 0.0047 * (model_size / 108000000) ** (-1 / 3)
+
+    save_interval = {
+        "1B": 2500,
+        "7B": 1000,
+    }.get(args.model, 5000)
+
+    distributed_strategy = {"7B": DistributedStrategy.fsdp}.get(args.model, DistributedStrategy.ddp)
+
+    return TrainConfig(
+        run_name=run_name,
+        seed=6198,
+        wandb=None if not args.wandb else WandbConfig(name=run_name, group=run_name, project="olmo-ladder"),
+        model=model_config,
+        ddp=DDPConfig(),  # defaults are fine
+        fsdp=FSDPConfig(
+            sharding_strategy=ShardingStrategy.SHARD_GRAD_OP,
+            wrapping_strategy=FSDPWrapStrategy.by_block_and_size,
+            precision=FSDPPrecision.mixed,
+        ),
+        optimizer=OptimizerConfig(
+            name=OptimizerType.adamw,
+            learning_rate=lr,
+            weight_decay=0.05,
+            eps=1e-8,
+            decay_norm_and_bias=True,
+            decay_embeddings=True,
+            betas=(0.9, 0.95),
+            metrics_log_interval=10,
+        ),
+        scheduler=SchedulerConfig(
+            name=SchedulerType.cosine_with_warmup,
+            alpha_f=0.01,
+            warmup_min_lr=0.0,
+            t_warmup=round(model_size / (global_batch_size * model_config.max_sequence_length)),
+        ),
+        max_duration=f"{length_in_tokens}T",
+        global_train_batch_size=global_batch_size,
+        tokenizer=TokenizerConfig(identifier="tokenizers/allenai_gpt-neox-olmo-dolma-v1_5.json"),
+        save_folder=f"runs/{run_name}",
```

Comment:
Put this on weka?

Reply:
I tried that first. Didn't work. Weka wasn't writeable. But also, it doesn't matter while the remote save folder is in s3. This folder is only temporary.

Reply:
Ok but it should be writable, so not sure what's going on there

Reply:
I haven't had an issue with my recent runs. They all write to weka.